src/Microsoft.Data/PrimitiveColumnContainer.cs

```diff
@@ -36,8 +43,10 @@ public PrimitiveColumnContainer(T[] values)
                 {
                     curBuffer = new DataFrameBuffer<T>();
                     Buffers.Add(curBuffer);
+                    NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
                 }
                 curBuffer.Append(values[i]);
+                SetBit(Length, true);
```

Comment:
Why are we doing this? It shouldn't be possible to get or set the null value at index `Length`. That should be out of range.

Reply:
Right, `GetValidityBit` does throw an exception when `index >= Length`. The set here is still valid because the current length is `Length + 1`, after the `curBuffer.Append` call just above. Line 50(the next line) is `Length++` and updates the length.